pkg/policy/mapstate.go

```diff
@@ -681,7 +689,7 @@ func (e *mapStateEntry) deepEqual(o *mapStateEntry) bool {
 	return true
 }
 
-func (e MapStateEntry) WithAuthType(authType AuthType) MapStateEntry {
+func (e MapStateEntry) withAuthType(authType AuthType) MapStateEntry {
```

Comment:
Does the "unused" lint failure here mean we don't have test cases covering this? Or just that we have created the entries another way?

This is a useful function regardless, feel free to add a `//nolint`

Reply:
We now have helpers like `withExplicitAuth()` and `withDerivedAuth()` so this was a real leftover, removed.